src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -1079,6 +1082,33 @@ private static string DetermineRuntimePropName(string clrPropName, string? jsonP
                 return runtimePropName;
             }
 
+            private static string DeterminePropNameVarName(string runtimePropName)
+            {
+                const string PropName = "PropName_";
+
+                // Use a different prefix to avoid possible collisions with "PropName_" in
```

Comment:
Perhaps add a TODO comment that the BASE64 approach is a temporary mitigation?

Reply:
Sure I'll add a comment.

It's not truly BASE64 since we only use standard hexadecimal 0-9 and A-F and not the full 64 characters.